bravado/testing/response_mocks.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+import mock
+
+from bravado.exception import BravadoTimeoutError
+from bravado.http_future import FALLBACK_EXCEPTIONS
+from bravado.response import BravadoResponseMetadata
+
+
+class BravadoResponseMock(object):
+    """Class that behaves like the :meth:`.HttpFuture.response` method as well as a :class:`.BravadoResponse`.
+    Please check the documentation for further information.
+    """
+
+    def __init__(self, result, metadata=None):
+        self._result = result
+        if metadata:
+            self._metadata = metadata
+        else:
+            self._metadata = BravadoResponseMetadata(
+                incoming_response=mock.Mock(
```

Comment:
I'm wondering if we can do something different to have a "mock" response without having mock library.
An example could be found on https://github.com/Yelp/bravado-core/blob/master/tests/profiling/unmarshal_response_profiler_test.py#L6

Doing this will prevent us to have the testing extra dependency.

In general that is not a problem, but I don't think will be un-realistic having `bravado[testing]==X.X.X` in the production requirements and it will add `mock` dependency in production.

Reply:
Good suggestion! I'm working that in.